drones/image_processing/yolo.py

```diff
@@ -0,0 +1,138 @@
+import time
+import typing
+import configparser
+import torch
+import torch.backends.cudnn as cudnn
+import numpy as np
+from yolov5.models.experimental import attempt_load
+from yolov5.utils.datasets import letterbox
+from yolov5.utils.general import check_img_size, non_max_suppression, scale_coords, xyxy2xywh
+from yolov5.utils.torch_utils import select_device
+import logging
+
+
+class YoloDetection:
+    def __init__(self):
+        # initialize config
+        self.config_parser = configparser.ConfigParser()
+        self.config_parser.read("image_processing/config.ini")
+        self.config = self.config_parser["YOLO"]
+
+        # Parse classes from config
+        config_classes = self.config["CLASSES"]
+        if config_classes != "None":
+            det_classes = config_classes.split(" ")
+            self.classes = []
+            for det_class in det_classes:
+                self.classes.append(int(det_class))
+        else:
+            self.classes = None  # type: ignore
+
+        self.device_type = self.config["DEVICE"]
+        self.model = attempt_load(self.config["NETWORK_PATH"], map_location=self.device_type)  # load FP32 model
+        self.stride = int(self.model.stride.max())  # model stride
+        self.conf_tres = float(self.config["CONFIDENCE_THRESHOLD"])
+        self.img_size = int(self.config["IMG_SIZE"])
+        self.iou_thres = float(self.config["IOU_THRESHOLD"])
+
+    def detect(self, img0: np.ndarray) -> list:
+        """Detect object on image using provided weights. Objects are detected by YOLO neural network
+
+        Parameters:
+        ----------
+            img0: np.ndarray
+                image on which objects detection will be proceeded
+        Returns:
+        ----------
+            It returns list of all detected objects on photo. Every element consist 5 values. First one is recognized
+            class name. Others are central position of object(x,y) and size of object(width, height).
+            All values are normalized in yolo norm.
+            To have position and size on original photo you have to multiply this results by original photo size.
+        """
+        # Load model
+
+        img_size = check_img_size(self.img_size, s=self.stride)  # check img_size
+        img = img0.copy()
+
+        img = letterbox(img, img_size, self.stride)[0]
+
+        # Convert
+        img = img[:, :, ::-1].transpose(2, 0, 1)  # BGR to RGB, to 3x416x416
+        img = np.ascontiguousarray(img)
+
+        # Get names and colors
+        names = self.model.module.names if hasattr(self.model, "module") else self.model.names
+
+        device = select_device(self.device_type)
+
+        # Run inference
+        if device.type != "cpu":
```

Comment:
What if device.type is not "cpu"? If we assume that model should not run, this config parameter is not required.

Reply:
it is configuration in case of using GPU